README.md

```diff
@@ -38,6 +38,38 @@ Each entry has the following values:
   - Written in present tense and describes the steps to take to meet the requirement
   - May outline recommendations, examples, or best practices
 
+## baseline-compiler
+
+The Security Baseline SIG maintains a Golang tool for validating and rendering Baseline controls. It includes the following commands:
+
+* [compile](#compile): Compile a YAML file of security controls
+* completion  Generate the autocompletion script for the specified shell
+* help        Help about any command
+* [oscal](#oscal): Write the Baseline definition to an OSCAL json catalog
+* [validate](#validate): Validate the baseline data files
+
+From the `cmd/` directory, run `go run . [command] [arguments]`
+
+### compile
+
+This command reads the yaml contents and generates two optional Markdown files: a listing of all OSPS Baseline controls for use in a static site generator and a checklist of controls by level.
+
+To produce a file for use in a static site generator (which is how we create baseline.openssf.org!), specify a file argument to `--output`.
+
+To produce a checklist file to use as an aid to evaluating your project's conformance to OSPS Baseline, specify a file argument to `--checklist-output`.
+
+Use the `--help` flag for more options.
+
+### oscal
+
+This command validates the OSPS Baseline output against [OSCAL](https://pages.nist.gov/OSCAL/). By default, it writes the JSON output to STDOUT. You can specify a file with `--out`. 
```

Comment:
I think it converts the catalog controls in the baseline from a SCI Layer 2 YAML format to an OSCAL JSON format. I read the code and I don't see any explicit validation of the OSCAL payload against any JSON schema, but that me be happening within the oscal module used.

@eddie-knight would know better, I suspect

Reply:
We have a presubmit that checks the oscal output:

https://github.com/ossf/security-baseline/actions/runs/14767602657/job/41461983532